### PR TITLE
rac2: add AdmittedTracker interface for use by RangeController

### PR DESCRIPTION
**Pull request by sumeerbhola**
The AdmittedTracker provides the latest AdmittedVector, and will be implemented by processorImpl.

Informs #129508

Epic: CRDB-37515

Release note: None

### CODE REVIEW

File: pkg/kv/kvserver/kvflowcontrol/rac2/range_controller_test.go [hunk 1]
```diff
@@ -107,8 +118,8 @@ func (r *testingRCRange) admit(
 	for _, replica := range r.mu.r.replicaSet {
 		if replica.desc.StoreID == storeID {
 			replica := replica
-			replica.info.Admitted[AdmissionToRaftPriority(pri)] = toIndex
-			replica.info.Term = term
+			replica.av.Admitted[AdmissionToRaftPriority(pri)] = toIndex
+			replica.av.Term = term
```
Comment (pav-kv) [inline pkg/kv/kvserver/kvflowcontrol/rac2/range_controller_test.go:122]: What happens with the priorities != `pri` when `Term` goes up? Should we rather pass the whole `AdmittedVector` in and update it wholesale?

**Review by pav-kv — APPROVED** [review]

**Review by sumeerbhola — COMMENTED** [review]
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130054)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @kvoli and @pav-kv)

___
*[`pkg/kv/kvserver/kvflowcontrol/rac2/range_controller_test.go` line 122 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130054#gh-1743292152:-O5xjeui35hAbQf1QR30:bnuvf5i) ([raw file](https://github.com/cockroachdb/cockroach/blob/39902be6419d2268069cff2563022d06b1fd742a/pkg/kv/kvserver/kvflowcontrol/rac2/range_controller_test.go#L122)):*
<details><summary><i>Previously, pav-kv (Pavel Kalinnikov) wrote…</i></summary><blockquote>

What happens with the priorities != `pri` when `Term` goes up? Should we rather pass the whole `AdmittedVector` in and update it wholesale?
</blockquote></details>

This was probably just a quick convenience in order to be able to provide just an incremental update in the test. I have added a TODO to improve this.

**Comment by sumeerbhola** [comment]
TFTR!

**Review by kvoli — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130054#-:-O5xkhjj9iWEbSkoMCeQ:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @pav-kv)

___
*[`pkg/kv/kvserver/kvflowcontrol/rac2/range_controller_test.go` line 122 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130054#gh-1743292152:-O5xkd7e8u0DMZ7-yRiz:bw4bg8n) ([raw file](https://github.com/cockroachdb/cockroach/blob/39902be6419d2268069cff2563022d06b1fd742a/pkg/kv/kvserver/kvflowcontrol/rac2/range_controller_test.go#L122)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

This was probably just a quick convenience in order to be able to provide just an incremental update in the test. I have added a TODO to improve this.
</blockquote></details>

It was convenience.  We can change it.

**Comment by sumeerbhola** [comment]
bors r=pav-kv,kvoli

**Pull request merged**